crates/ide/src/lib.rs

```diff
@@ -479,7 +479,7 @@ impl Analysis {
         &self,
         position: FilePosition,
         new_name: &str,
-    ) -> Cancelable<Option<RangeInfo<SourceChange>>> {
+    ) -> Cancelable<Result<Option<RangeInfo<SourceChange>>, Box<dyn Error + Send + Sync>>> {
```

Comment:
I think the right signature here should be 

```rust
Cancelable<Result<RangeInfo<SourceChange>, RenameError>>
```

that is,

* use concrete type for an error
* fold `None` into one variant of `Error`

Reply:
I've applied your proposal and turned `None` into errors. A few open questions remain, though. I left corresponding TODO comments in the latest commit.